optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/solver/WannabeSolverManager.java

```diff
@@ -108,4 +115,40 @@ public Roster getRoster(final Integer tenantId) {
         Solver<Roster> solver = tenantIdToSolverMap.get(tenantId);
         return solver == null ? null : solver.getBestSolution();
     }
+
+    public Map<Shift, IndictmentView> getShiftIndictmentMap(Integer tenantId) {
+        if (tenantIdToSolverMap.containsKey(tenantId)) {
+            Roster roster = tenantIdToSolverMap.get(tenantId).getBestSolution();
+            ScoreDirector<Roster> scoreDirector = tenantIdToSolverMap.get(tenantId).getScoreDirectorFactory()
+                    .buildScoreDirector();
+            scoreDirector.setWorkingSolution(roster);
+            Map<Shift, IndictmentView> indictmentMap = tenantIdToShiftIndictmentMap.computeIfAbsent(tenantId, (k) -> new HashMap<>());
+            indictmentMap.clear();
+            scoreDirector.getIndictmentMap().forEach((k, v) -> {
+                if (k instanceof Shift) {
+                    indictmentMap.put((Shift) k, new IndictmentView(v));
+                }
+            });
+            return indictmentMap;
+        }
+        return tenantIdToShiftIndictmentMap.getOrDefault(tenantId, Collections.emptyMap());
```

Comment:
So, if we can't find it, there we no indictments?
This is "let's sweep the problem under the rug" code. The UI then incorrectly reports that the schedule is perfect (while it is not).

If it can't find it, it must fail fast. It must stick out like soar thumb. And if that's an ugly red popup in the UI, that's a good thing for us. Much better than an "all is fine" look with invalid data. Then we either fix it properly or we figure out a way to notify the user in the UI in a pretty way (think notification) that we don't know.